lib/cli-manager.js

```diff
@@ -0,0 +1,352 @@
+      this._notifier.notify({ defer: false });
+    }
+  }
+
+  _authIsRequired(cmd) {
+    if (AllCommandsNotRequiringAuth.includes(cmd)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  // TODO: refactor the auth stuff or at least document it :(
+  _processAuthOptions(options) {
+    const commandName = utils.getCommandNameFromOptions(options);
+    this._requiresAuth = this._authIsRequired(commandName);
+
+    const disregardOptions = commandName === 'init';
+
+    // clear all auth options
+    if (disregardOptions) {
+      Object.keys(AuthOptionsNames).forEach((key) => {
+        const propName = AuthOptionsNames[key];
+        options[propName] = null;
+      });
+
+      return;
+    }
+
+    const isOneTimeSession = !options[AuthOptionsNames.PROFILE]
+      && (options[AuthOptionsNames.EMAIL] && options[AuthOptionsNames.PASSWORD]);
+
+    if (isOneTimeSession) {
+      const isValidEmail = utils.isValidEmail(options[AuthOptionsNames.EMAIL]);
+      if (!isValidEmail) {
+        throw new KinveyError(Errors.InvalidEmail);
+      }
+
+      if (options[AuthOptionsNames.HOST]) {
+        options[AuthOptionsNames.HOST] = utils.formatHost(options[AuthOptionsNames.HOST]);
+      }
+
+      this._isOneTimeSession = true;
+    }
+  }
+
+  /**
+   * Tries to set the current user from a profile name or from the active profile. If there are no saved profiles, it doesn't set it.
+   * If no name is provided, no active profile is set and there is only one saved profile, then it sets the user to this profile.
+   * @param {String} [profileName] Name of already saved profile. Throws if not found.
+   * @private
+   */
+  _setCurrentUserFromProfile(profileName) {
+    debugger;
+    if (this._isOneTimeSession || !this._setup.hasProfiles()) {
+      return;
+    }
+
+    let profile;
+    if (profileName) {
+      profile = this._setup.findProfileByName(profileName);
+    } else if (this._setup.hasActiveProfile()) {
+      profile = this._setup.getActiveProfile();
+    } else {
+      // profile is not provided explicitly, active profile is not set
+      const allProfiles = this._setup.getProfiles();
+      const isSingleProfile = Object.keys(allProfiles).length === 1;
+      if (isSingleProfile) {
+        const singleProfileName = Object.keys(allProfiles)[0];
+        profile = this._setup.findProfileByName(singleProfileName);
+      }
+    }
+
+    if (!profile && profileName) {
+      throw new KinveyError(Errors.ProfileNotFound);
+    }
+
+    if (profile) {
+      this._authentication.setCurrentUser(profile);
+    }
+  }
+
+  _loadGlobalSetup() {
+    if (this._isOneTimeSession) {
+      return;
+    }
+
+    const err = this._setup.load();
+    const canContinue = !err || err instanceof Error && err.code === 'ENOENT';
+    if (!canContinue) {
+      this._logger.error('Failed to load global settings.');
+      throw err;
+    }
+  }
+
+  _initCommandsManager(args) {
+    if (args) {
```

Comment:
Per the style guide (https://github.com/Kinvey/style-guides/blob/master/backend/es6.md#comparison--shortcuts), shortcuts should only be used for `boolean` values. Should be `if (args != null)` here and for all other `null`/`undefined` checks

Reply:
Thanks for the link. I really have missed this paragraph   :thumbsup: